File: reto7.py
#Enunciado: Crea un programa que cuente cuantas veces se repite cada palabra y que muestre el 
# recuento final de todas ellas. - Los signos de puntuación no forman parte de la palabra. 
# - Una palabra es la misma aunque aparezca en mayúsculas y minúsculas. 
# - No se pueden utilizar funciones propias del lenguaje que lo resuelvan automáticamente.


texto = "hola, como estas?. hola que onda? que como"
for i in (("?", ""), (".", ""),(",","")):
    texto = texto.replace(*i)
texto = texto.lower()
texto = (texto.split(" "))



repetida = {}
cont = 0
for i in texto:
    if cont == 0:     
        repetida[i] = 1
    else:
        if  not any(i == clave for clave,valor in repetida.items()):
            repetida[i] = 1
        else:
            repetida[i] +=1
    cont +=1


for clave,valor in repetida.items():
    print(f"hay {valor} palabra {clave} ")

        

File: reto7.js
let texto = "Hola, como estas?. Hola que onda? que como"
texto = texto.replace(/[?]/g,"").replace(/[,]/g,"").replace(/[.]/g,"").toLowerCase().split(" ")

let repetida = {}
let cont = 0
for (let i of texto) {
    if (cont == 0){
        repetida[i]=1
    }else{
        if (!(Object.entries(repetida).some(j=> j[0]==i))) {
            repetida[i]=1
        } else {
            repetida[i] +=1
        }
    }
    cont +=1
}

Object.entries(repetida).forEach(([clave,valor])=>{
    console.log("hay "+valor+" palabra "+clave)})